govcd/defined_entity.go

```diff
@@ -204,6 +205,163 @@ func (rdeType *DefinedEntityType) Delete() error {
 	return nil
 }
 
+// GetAllBehaviors retrieves all the Behaviors of the receiver RDE Type. Query parameters can be supplied to modify pagination.
+func (rdeType *DefinedEntityType) GetAllBehaviors(queryParameters url.Values) ([]*types.Behavior, error) {
+	if rdeType.DefinedEntityType.ID == "" {
+		return nil, fmt.Errorf("ID of the receiver Defined Entity Type is empty")
+	}
+	return getAllBehaviors(rdeType.client, rdeType.DefinedEntityType.ID, types.OpenApiEndpointRdeTypeBehaviors, queryParameters)
+}
+
+// GetBehaviorById retrieves a unique Behavior that belongs to the receiver RDE Type and is determined by the
+// input ID. The ID can be a RDE Interface Behavior ID or a RDE Type overridden Behavior ID.
+func (rdeType *DefinedEntityType) GetBehaviorById(id string) (*types.Behavior, error) {
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointRdeTypeBehaviors
+	apiVersion, err := rdeType.client.getOpenApiHighestElevatedVersion(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := rdeType.client.OpenApiBuildEndpoint(fmt.Sprintf(endpoint, rdeType.DefinedEntityType.ID), id)
+	if err != nil {
+		return nil, err
+	}
+
+	response := types.Behavior{}
+	err = rdeType.client.OpenApiGetItem(apiVersion, urlRef, nil, &response, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return &response, nil
+}
+
+// GetBehaviorByName retrieves a unique Behavior that belongs to the receiver RDE Type and is named after
+// the input.
+func (rdeType *DefinedEntityType) GetBehaviorByName(name string) (*types.Behavior, error) {
+	behaviors, err := rdeType.GetAllBehaviors(nil)
+	if err != nil {
+		return nil, fmt.Errorf("could not get the Behaviors of the Defined Entity Type with ID '%s': %s", rdeType.DefinedEntityType.ID, err)
+	}
+	for _, b := range behaviors {
+		if b.Name == name {
+			return b, nil
+		}
+	}
```

Comment:
I think it is not, because Behaviors are named only when created in Interfaces (you cannot change the name when overriding in a RDE Type), and the name is unique per interface.
But let me double check and I'll come back with a confirmation.

Reply:
Checked. I also added an assert in `defined_interface_test.go:167` to check what happens if one wants to create another Behavior with same name (for the same interface). It returns RDE_BEHAVIOR_ALREADY_EXISTS.
So Type behaviors will behave the same, as they can't create but just override. And if it is overridden, it replaces the original, so it won't have 2 but 1.

Reply:
Thanks!